projects/control-service/projects/pipelines_control_service/src/integration-test/java/com/vmware/taurus/datajobs/it/DataJobOOMIT.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2021 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.vmware.taurus.datajobs.it;
+
+import com.vmware.taurus.ControlplaneApplication;
+import com.vmware.taurus.controlplane.model.data.DataJobExecution;
+import com.vmware.taurus.datajobs.it.common.BaseIT;
+import com.vmware.taurus.datajobs.it.common.DataJobDeploymentExtension;
+import com.vmware.taurus.datajobs.it.common.JobExecutionUtil;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.springframework.boot.test.context.SpringBootTest;
+
+
+@Slf4j
+@SpringBootTest(
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
+    classes = ControlplaneApplication.class)
+public class DataJobOOMIT extends BaseIT {
+
+  @RegisterExtension
+  static DataJobDeploymentExtension dataJobDeploymentExtension =
+      new DataJobDeploymentExtension("oom_job.zip");
+
+  @Test
+  public void testDataJob_causesOOM_shouldCompleteWithUserError(
```

Comment:
Did you verify the test fails without the fix (the change in KuberntesService). 

I am wondering - 
If the error happens after the job is finished. Shouldn't we wait for some condition for the status to change ?

Reply:
Yes, I did with another JobExecutionUtil.checkDataJobExecutionStatus(
                executionId,
                DataJobExecution.StatusEnum.PLATFORM_ERROR,
                opId,
                jobName,
                teamName,
                username,
                mockMvc);

However, it takes more than 5 minutes and it's not consistent.